File: backend/src/QuinntyneBrownStudio.Domain/Models/QuoteResult.cs
using QuinntyneBrownStudio.Domain.Entities;
using QuinntyneBrownStudio.Domain.ValueObjects;

namespace QuinntyneBrownStudio.Domain.Models;

public sealed record QuoteResult(
    long InputRevision,
    long ConfigurationRevision,
    QuoteLine[] Lines,
    Money Subtotal,
    AppliedDiscount Discount,
    Money Total,
    AvailabilityResult Availability
);

File: frontend/projects/domain/src/lib/quote-result.ts
import { Money } from './money';
export interface QuoteResult {
  inputRevision: number;
  configurationRevision: number;
  lines: {
    kind: string;
    quantity: string;
    amount: Money;
    locationIndex: number | null;
  }[];
  subtotal: Money;
  discount: {
    percentage: string;
    amount: Money;
    kind: string | null;
    codeError: string | null;
  };
  total: Money;
  availability: {
    available: boolean;
    photographerIds: string[];
    reasonCode: string | null;
  };
}
